hoomd/md/pytest/test_update_group_dof.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import hoomd
+import pytest
+
+
+@pytest.fixture(scope='session')
+def snapshot_factory(device):
+    """Make a test snapshot for update_group_dof testing."""
+
+    def make_snapshot():
+        """Make the snapshot."""
+        snap = hoomd.Snapshot(device.communicator)
+
+        if snap.communicator.rank == 0:
+            box = [10, 10, 10, 0, 0, 0]
+            snap.configuration.box = box
+            snap.particles.N = 3
+            snap.particles.position[:] = [[0, 1, 0], [-1, 1, 0], [1, 1, 0]]
+            snap.particles.velocity[:] = [[0, 0, 0], [0, -1, 0], [0, 1, 0]]
+            snap.particles.moment_inertia[:] = [
+                [2.0, 0, 0],
+                [1, 1, 1],
+                [1, 1, 1],
+            ]
+            snap.particles.angmom[:] = [[0, 2, 4, 6]] * 3
+            snap.particles.types = ['A', 'B']
+
+        return snap
+
+    return make_snapshot
+
+
+def test_set_snapshot(simulation_factory, snapshot_factory):
+    """Test that the number of degrees of freedom updates after set_snapshot."""
+    snapshot = snapshot_factory()
+
+    sim = simulation_factory(snapshot)
+    filter_all = hoomd.filter.All()
+    method = hoomd.md.methods.Langevin(filter=filter_all, kT=1.0)
+    sim.operations.integrator = hoomd.md.Integrator(
+        0.005,
+        methods=[method],
+        integrate_rotational_dof=True,
+    )
+    thermo = hoomd.md.compute.ThermodynamicQuantities(filter=filter_all)
+    sim.operations.add(thermo)
+
+    # check initial degrees of freedom in snapshot
+    sim.run(0)
+    assert thermo.translational_degrees_of_freedom == 9
+    assert thermo.rotational_degrees_of_freedom == 7
+
+    # add a particle and check after set_snapshot
+    if snapshot.communicator.rank == 0:
+        snapshot.particles.N = 4
+        if snapshot.communicator.rank == 0:
+            snapshot.particles.moment_inertia[3] = [0, 0, 1]
```

Comment:
```suggestion
        snapshot.particles.moment_inertia[3] = [0, 0, 1]
```
Since you already know you're on rank 0 at this point, the extra check shouldn't be necessary